src/Model/lbsProcessor.py

```diff
@@ -82,14 +84,38 @@ def get_objects(self):
         """Return objects assigned to processor
         """
 
+        return self.objects.union([self.sentinel])
+
+    def get_migratable_objects(self):
+        """Return migratable objects assigned to processor
+        """
+
         return self.objects
 
+    def get_sentinel_object(self):
+        """Return sentinel object assigned to processor
+        """
+
+        return self.sentinel
+
     def get_object_ids(self):
         """Return IDs of objects assigned to processor
         """
 
+        return [o.get_id() for o in self.objects + [self.sentinel]]
```

Comment:
`Class 'set' does not define '__add__', so the '+' operator cannot be used on its instances `

I am not sure what's the intention here.
If you want to return a `list` with all objects ids including sentinel it should be:
```
obj_ids_list = [o.get_id() for o in self.objects]
obj_ids_list.append(self.sentinel.get_id())
return obj_ids_list
```
or
```
obj_ids_list = [o.get_id() for o in self.objects]
obj_ids_list.append(self.sentinel.index)
return obj_ids_list
```
If you want to return a `list` with objects ids and sentinel id as another `list` inside the first one it should be:
```
obj_ids_list = [o.get_id() for o in self.objects]
obj_ids_list.append([self.sentinel.get_id()])
return obj_ids_list
```
or
```
obj_ids_list = [o.get_id() for o in self.objects]
obj_ids_list.append([self.sentinel.index]`)
return obj_ids_list
```

Reply:
Oh yes, thanks for pointing that, I erroneously treated the objects as `list` whereas it is a `set`, so as to not allow for element replication.